test/types/tsconfig.json

```diff
@@ -3,7 +3,7 @@
     "outDir": "../../build/tmp",
     "sourceMap": true,
     "declaration": false,
-    "moduleResolution": "node",
+    "moduleResolution": "Node10",
```

Comment:
Why this change, since the commit message contains no information about this?

Reply:
Moreover, https://www.typescriptlang.org/tsconfig#moduleResolution indicates we shouldn't rely on this.

Reply:
I have tried that. See https://github.com/mozilla/pdf.js/compare/master...tamuratak:pdf.js:better_types_c

With the `nodenext` option, we have to add the `.js` extension to every `import("./filename")` on JSDoc. The difficult part is that making `tsc-alias` do the job when it rewrites import statements. I had to switch to `tsconfig-replace-paths` instead of `tsc-alias`. 

https://www.typescriptlang.org/docs/handbook/modules/theory.html#module-resolution-is-host-defined

> In Node.js, module specifiers in import statements and dynamic import() calls are not allowed to omit file extensions or /index.js suffixes, while module specifiers in require calls are. This module resolution mode understands and enforces this restriction

Reply:
OK, let's leave that as-is for this patch then.